plugins/blockvolume/api/types.go

```diff
@@ -0,0 +1,38 @@
+package api
+
+// BlockVolumeInfo represents block volume info
+type BlockVolumeInfo struct {
+	// HostingVolume name is optional
+	HostingVolume string `json:"hosting_volume"`
+	// Name represents block Volume name
+	Name string `json:"name"`
+	// Size represents Block Volume size in bytes
+	Size    int64 `json:"size,omitempty"`
+	HaCount int   `json:"hacount,omitempty"`
+}
+
+// BlockVolumeCreateRequest represents req Body for Block vol create req
+type BlockVolumeCreateRequest struct {
+	*BlockVolumeInfo
+	Clusters []string `json:"clusters,omitempty"`
+	Auth     bool     `json:"auth,omitempty"`
+}
+
+// BlockVolumeCreateResp represents resp body for a Block Vol Create req
+type BlockVolumeCreateResp struct {
+	*BlockVolumeInfo
```

Comment:
as you are using `BlockVolumeInfo` `size` cannot be omitempty for this volume response

Reply:
Since we are getting only list of blockvolume name from `gluster-block list` command.
So for a blockvolume list response, we are only showing blockvolume name and hosting volume name, so if I remove `omitempty` then size will be displayed as 0 in response.

Reply:
for a BlockVolumeList response `size` is not a mandatory field